diff --git a/vllm/v1/attention/backends/mla/prefill/flashinfer.py b/vllm/v1/attention/backends/mla/prefill/flashinfer.py
index a4214d1..2eaf4a9 100644
--- a/vllm/v1/attention/backends/mla/prefill/flashinfer.py
+++ b/vllm/v1/attention/backends/mla/prefill/flashinfer.py
@@ -42,6 +42,11 @@ class FlashInferPrefillBackend(MLAPrefillBackend):
             qk_rope_head_dim=64,
             v_head_dim=128,
         ),
+        MLADimensions(
+            qk_nope_head_dim=192,
+            qk_rope_head_dim=64,
+            v_head_dim=256,
+        ),
     ]
 
     @staticmethod
@@ -50,7 +55,7 @@ class FlashInferPrefillBackend(MLAPrefillBackend):
 
     @classmethod
     def supports_compute_capability(cls, device_capability: "DeviceCapability") -> bool:
-        return device_capability.major == 10
+        return device_capability.major in (10, 12)
 
     @classmethod
     def is_available(cls) -> bool:
@@ -99,7 +104,10 @@ def _ensure_chunks(
             for _ in range(len(self._prefill_chunks), num_chunks):
                 self._prefill_chunks.append(
                     BatchPrefillWithRaggedKVCacheWrapper(
-                        workspace_buffer, "NHD", backend="cutlass"
+                        workspace_buffer,
+                        "NHD",
+                        backend="fa2"
+                        if torch.cuda.get_device_capability()[0] == 12 else "cutlass",
                     )
                 )
 
@@ -144,7 +152,10 @@ def prepare_metadata(
         if self._prefill_main is None:
             self._prefill_main = BatchPrefillWithRaggedKVCacheWrapper(
-                self._workspace_buffer, "NHD", backend="cutlass"
+                self._workspace_buffer,
+                "NHD",
+                backend="fa2"
+                if torch.cuda.get_device_capability()[0] == 12 else "cutlass",
             )
             self._ensure_chunks(_DEFAULT_NUM_CHUNKS, self._workspace_buffer)
